internal/impl/docker.go

```diff
@@ -108,70 +114,55 @@ downloaded and installed in the container. Do you want to proceed? [Y/n] `)
-			return nil, fmt.Errorf("user bailed out")
+			return "", fmt.Errorf("user bailed out")
 		}
-		toInstall = append(toInstall, "github.com/ServiceWeaver/weaver-kube/cmd/weaver-kube@latest")
+		install = "github.com/ServiceWeaver/weaver-kube/cmd/weaver-kube@latest"
 	} else {
-		// Released tool binary that's not compiled to linux/amd64. Re-install
-		// it inside the container.
-		toInstall = append(toInstall, "github.com/ServiceWeaver/weaver-kube/cmd/weaver-kube@"+toolVersion)
-	}
-
-	if image == "" {
-		image = fmt.Sprintf("%s:%s", app.Name, depId[:8])
-	}
-
-	return &buildSpec{
-		image:     image,
-		files:     toCopy,
-		goInstall: toInstall,
-	}, nil
-}
-
-// buildImage builds a docker image with a given spec.
-func buildImage(ctx context.Context, spec *buildSpec) error {
-	fmt.Fprintf(os.Stderr, greenText(), fmt.Sprintf("Building image %s...", spec.image))
-	// Create:
-	//  workDir/
-	//    file1
-	//    file2
-	//    ...
-	//    fileN
-	//    Dockerfile   - docker build instructions
-	ctx, cancel := context.WithTimeout(ctx, time.Second*120)
-	defer cancel()
-
-	// Create workDir/.
-	workDir := filepath.Join(os.TempDir(), fmt.Sprintf("weaver%s", uuid.New().String()))
-	if err := os.Mkdir(workDir, 0o700); err != nil {
-		return err
+		// Install the currently running version of "weaver-kube" in the
+		// container.
+		install = "github.com/ServiceWeaver/weaver-kube/cmd/weaver-kube@" + toolVersion
 	}
-	defer os.RemoveAll(workDir)
 
-	// Copy the files from spec.files to workDir/.
-	for _, file := range spec.files {
-		workDirFile := filepath.Join(workDir, filepath.Base(filepath.Clean(file)))
-		if err := cp(file, workDirFile); err != nil {
-			return err
-		}
+	// Create a Dockerfile.
+	type content struct {
+		Install    string // "weaver-kube" binary to install, if any
+		Entrypoint string // container entrypoint
 	}
+	var template = template.Must(template.New("Dockerfile").Parse(`
+{{if .Install }}
+FROM golang:bullseye as builder
+RUN go install "{{.Install}}"
+{{end}}
 
-	// Create a Dockerfile in workDir/.
-	dockerFile, err := os.Create(filepath.Join(workDir, dockerfileTmpl.Name()))
+FROM ubuntu:rolling
+WORKDIR /weaver/
+COPY . .
+{{if .Install }}
+COPY --from=builder /go/bin/ /weaver/
+{{end}}
+ENTRYPOINT ["{{.Entrypoint}}"]
+`))
+	dockerFile, err := os.Create(filepath.Join(workDir, "Dockerfile"))
 	if err != nil {
-		return err
+		return "", err
 	}
-	if err := dockerfileTmpl.Execute(dockerFile, spec.goInstall); err != nil {
-		dockerFile.Close()
-		return err
+	defer dockerFile.Close()
+	c := content{Install: install}
+	if install != "" {
+		c.Entrypoint = "/weaver/weaver-kube"
+	} else {
+		c.Entrypoint = filepath.Join("/weaver", filepath.Base(tool))
 	}
-	if err := dockerFile.Close(); err != nil {
-		return err
+	if err := template.Execute(dockerFile, c); err != nil {
+		return "", err
 	}
-	return dockerBuild(ctx, workDir, spec.image)
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*120)
```

Comment:
Should we make `time.Second * 120` a constant and put it at the top of this file? We may want to change it (eventually let the user control it).

Reply:
Done.